cookbooks/bcpc-hadoop/libraries/helpers.rb

```diff
@@ -0,0 +1,66 @@
+
+module Bcpc_Hadoop
+  module Helper
+
+    include Chef::Mixin::ShellOut
+
+    # Apply translation rules for Hortonworks release strings
+    # to the debian package name strings
+    #
+    # release - String dotted Hortonworks release (e.g. 2.2.0-2041)
+    # package name - String for base package name
+    #
+    # Raises RuntimeError on any unspecified error
+    # Returns - string to use in a Hortonworks debian package name
+    # (i.e. 2.2.0.2041 for a release and package like
+    #  hadoop-hdfs-datanode to something akin to
+    #  hadoop-2-2-0-2041-hdfs-datanode)
```

Comment:
Is this a standard convention or just something we have seen so far. In any case, having a library certainly makes change easy.